### ISSUE TITLE
Brightness sync between external monitor and laptop monitor incorrect after disconnecting monitor and changing brightness.

### ISSUE THREAD

**Issue by lbrtdy** [emoacht/Monitorian#271]
I dock my laptop to a docking station on which an external monitor is attached along with a keyboard and mouse. The external monitor's brightness is usually 10 steps less than the laptop monitor's brightness. Eg if laptop monitor is set to '20', the external monitor is set to '10'. Sync is enabled to keep these two values tied together. Problems arise when I undock the laptop and change the laptop monitor brightness in this undocked state. Upon redocking, the sync is still enabled, but the step difference is no longer correct.

**Steps to reproduce**

1. Set laptop monitor brightness to '20'
2. Set external monitor brightness to '10'
3. Enable sync
4. Disconnect external monitor/undock laptop
5. Change laptop monitor brightness to '30'
6. Reconnect external monitor
7. Note that the external monitor kept it's brightness state at '10', but now the laptop monitor's brightness is at '30'. This is a difference of 20 instead of the original 10.
8. Increase brightness +1 via Monitorian slider. Note that the sync is kept and both monitors increase brightness by +1. External monitor is now at '11', laptop monitor is now at '31'.

**Expected behavior**
Not quite sure what the best behavior is here. Possible options would be to keep the 'high' value (laptop monitor's '30') locked until the external monitor's low value catches up to the original sync difference (until external monitor reaches '20') when adding brightness. Once the original sync value is reached, then they both adjust in unison. This behaves like the existing behavior where you sync 2 monitors and bring them both to 0. Upon adjusting the brightness back up up, the lower value monitor doesn't start changing brightness until the higher value monitor reaches the previous sync point. With this method the brightness of both monitors will be out of sync until manual intervention. It may require a lot of brightness button presses on my keyboard to get back in sync depending on how far they drifted apart.

Option number two would be to detect laptop dock/monitor attach event and re-send the appropriate value corresponding to the current laptop monitor's brightness value. Eg. a brightness of '20' will be sent to the external display because at the time of docking, the laptop monitor is set to '30'.

**Comment by emoacht** [comment]
Thanks for the report.
Could you post [operation.log](https://github.com/emoacht/Monitorian#operations)?

**Comment by lbrtdy** [comment]
[operation.log](https://github.com/emoacht/Monitorian/files/7758610/operation.log)

Thank you.

**Comment by emoacht** [comment]
Thanks for the log.
That behavior is completely normal but I got your point.

I improved restore function in Microsoft Store version to take into account monitors' unison state.
As of Ver 3.8.2, if a monitor is added to the monitors in unison and dettached, and if the brightness of another monitor in unison is changed, when the first monitor is reattached, its brightness will be adjusted as if it had been continuously attached. This is the change of restore function but I do not think it would be contrary to user expectations for this function. This behavior is similar to your option two.

To use this function, be sure to make `Restore set brightness` in the menu checked.
